tests/job_tests/corpus/test_transform.py

```diff
@@ -0,0 +1,257 @@
+    - The nested subcorpus sc2/sc1 shouldn't be merged with the main subcorpus sc1.
+    Moreover, its recording sc2/sc1/rec1 shouldn't be merged with neither rec1 nor sc1/rec1.
+    `MergeStrategy.MERGE_RECURSIVE` only takes into consideration the top level subcorpora.
+    - It's intended that sc3/rec2 doesn't have any segments.
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # First corpus.
+        c1_path = os.path.join(tmpdir, "c1.xml.gz")
+        c1_structure = {
+            "name": "c1",
+            "recordings": [
+                {"name": "rec1", "segments": ["seg10"]},
+            ],
+            "subcorpora": [
+                {
+                    "name": "sc1",
+                    "recordings": [
+                        {"name": "rec1", "segments": ["seg1", "seg2"]},
+                        {"name": "rec2", "segments": ["seg3", "seg4"]},
+                    ],
+                },
+                {
+                    "name": "sc2",
+                    "recordings": [
+                        {"name": "rec2", "segments": ["seg5"]},
+                        {"name": "rec3", "segments": ["seg6"]},
+                    ],
+                },
+            ],
+        }
+        c1 = CorpusCreatorHelper._create_corpus_with_structure(c1_structure)
+        c1.dump(c1_path)
+
+        # Second corpus.
+        c2_path = os.path.join(tmpdir, "c2.xml.gz")
+        c2_structure = {
+            "name": "c2",
+            "recordings": [
+                {"name": "rec1", "segments": ["seg13"]},
+            ],
+            "subcorpora": [
+                {"name": "sc1", "recordings": [{"name": "rec1", "segments": ["seg3", "seg4", "seg5"]}]},
+                {
+                    "name": "sc2",
+                    "subcorpora": [
+                        {
+                            "name": "sc1",
+                            "recordings": [{"name": "rec1", "segments": ["seg1"]}],
+                        }
+                    ],
+                },
+                {
+                    "name": "sc3",
+                    "recordings": [
+                        {"name": "rec1", "segments": ["seg1"]},
+                        {"name": "rec2"},
+                    ],
+                },
+            ],
+        }
+        c2 = CorpusCreatorHelper._create_corpus_with_structure(c2_structure)
+        c2.dump(c2_path)
+
+        # Merge the first and second corpora.
+        merged_corpus_path = os.path.join(tmpdir, "merged_corpus.xml.gz")
+        merge_corpora_job = MergeCorporaJob(
+            bliss_corpora=[Path(c1_path), Path(c2_path)],
+            name="custom_name_for_merged_corpus",
+            merge_strategy=MergeStrategy.MERGE_RECURSIVE,
+        )
+        merge_corpora_job.out_merged_corpus = Path(merged_corpus_path)
+        merge_corpora_job.run()
+
+        merged_corpus = libcorpus.Corpus()
+        merged_corpus.load(merged_corpus_path)
+        expected_merged_corpus_structure = {
```

Comment:
since you already have everything written up (Good job writing a test case 👍 ) would it be easy to just add the expected structures for the other merging methods as well and have full coverage of the job?

Reply:
Done :)